generators/server/templates/src/main/kotlin/package/security/oauth2/AuthorizationHeaderUtil.kt.ejs

```diff
@@ -18,31 +18,140 @@
+                val tokenValue = authentication.token.tokenValue
+                return Optional.of("${OAuth2AccessToken.TokenType.BEARER.value} $tokenValue")
+            }
+        }
+        return Optional.empty()
+    }
+
+    private fun refreshToken(client: OAuth2AuthorizedClient, oauthToken: OAuth2AuthenticationToken): String? {
+        val atr = refreshTokenClient(client)
+        if (atr?.accessToken == null) {
+            log.info("Failed to refresh token for user")
+            return null
+        }
+
+        val refreshToken = atr.refreshToken?: client.refreshToken
+        val updatedClient = OAuth2AuthorizedClient(
+            client.clientRegistration,
+            client.principalName,
+            atr.accessToken,
+            refreshToken
+        )
+
+        clientService.saveAuthorizedClient(updatedClient, oauthToken)
+        return atr.accessToken.tokenValue
+
+    }
+
+    private fun refreshTokenClient(currentClient: OAuth2AuthorizedClient): OAuth2AccessTokenResponse? {
+        val formParameters = LinkedMultiValueMap<String, String>()
+        formParameters.add(OAuth2ParameterNames.GRANT_TYPE, AuthorizationGrantType.REFRESH_TOKEN.value)
+        formParameters.add(OAuth2ParameterNames.REFRESH_TOKEN, currentClient.refreshToken.tokenValue)
+        formParameters.add(OAuth2ParameterNames.CLIENT_ID, currentClient.clientRegistration.clientId)
+        val requestEntity = RequestEntity
+            .post(URI.create(currentClient.clientRegistration.providerDetails.tokenUri))
+            .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+            .body(formParameters)
+        try {
+            val r = restTemplate(currentClient.clientRegistration.clientId, currentClient.clientRegistration.clientSecret)
+            val responseEntity = r.exchange(requestEntity, OAuthIdpTokenResponseDTO::class.java)
+            return toOAuth2AccessTokenResponse(responseEntity.body)
+        } catch (e: OAuth2AuthorizationException) {
+            log.error("Unable to refresh token", e)
+            throw OAuth2AuthenticationException(e.error, e)
         }
 
-        return if (accessToken == null) null else "Bearer $accessToken"
+    }
+
+    private fun toOAuth2AccessTokenResponse(oAuthIdpResponse: OAuthIdpTokenResponseDTO): OAuth2AccessTokenResponse {
+        val additionalParameters = hashMapOf<String, Any>()
+        additionalParameters["id_token"] = oAuthIdpResponse.idToken
+        additionalParameters["not-before-policy"] = oAuthIdpResponse.notBefore
+        additionalParameters["refresh_expires_in"] = oAuthIdpResponse.refreshExpiresIn
+        additionalParameters["session_state"] = oAuthIdpResponse.sessionState
+
+        return OAuth2AccessTokenResponse.withToken(oAuthIdpResponse.accessToken)
+            .expiresIn(oAuthIdpResponse.expiresIn)
+            .refreshToken(oAuthIdpResponse.refreshToken)
+            .scopes(Pattern.compile("\\s").split(oAuthIdpResponse.scope).toHashSet())
+            .build()
```

Comment:
You missed adding `additionalParameters` to the response, i.e `.additionalParameters(additionalParameters)`